pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -55,11 +55,12 @@ def __init__(self, trainer: "pl.Trainer", resume_from_checkpoint: Optional[_PATH
 
     @property
     def _hpc_resume_path(self) -> Optional[str]:
-        weights_save_path = self.trainer.weights_save_path
-        fs = get_filesystem(weights_save_path)
-        if not fs.isdir(weights_save_path):
+        # in v1.8 set this equal to self.trainer.default_root_dir
+        dir_path_hpc = self.trainer.weights_save_path
```

Comment:
Can you also update all the references of `trainer.weights_save_path` in the tests (even though you don't raise a warning)?

Reply:
> so we aren't emitting a warning when someone calls the getter for weights_save_path? Because we want to avoid causing a warning when using it internally?

that's correct. 

> Can you also update all the references of trainer.weights_save_path in the tests (even though you don't raise a warning)?

The only references of trainer.weights_save_path in the tests are in `loggers/test_all.py` which I believe are all necessary to test that weights_save_path is still working as intended until it is removed.

Reply:
@daniellepintz we could define a property & setter for weights save path on the trainer which emit a deprecation warning.

And usages within the framework can rely on a private attribute which backs the property's implementation, so no deprecation warnings are emitted then.

Reply:
Yes agreed, if it's possible I would definitely do that. The benefits of removing all deprecated references and re-writing the tests in the deprecation PR:
- The follow up removal in v1.8 is dead simple and can be done by anyone (good first issue)
- We can be certain already now that the removal will not need additional fixing in 1.8, because we adjust the tests now.

I think we want to follow this as a general rule for all removals.

Reply:
> we should define a property & setter for weights save path

@ananthsub why a setter? we don't have one now?

Reply:
Ok, I added a `_weights_save_path_internal` property